### PR TITLE
misc integration stability fixes

### PR DESCRIPTION
**Pull request by mchalski**
_tagging a lot of people - do a review or treat it as a fyi, whichever you prefer_ 

---

this work was triggered by https://tracker.mender.io/browse/MEN-4412, but:
- the first 2 commits are general must-haves that slipped my mind initially
- the last one is the fix for demo artifact test

after a couple amends - I got 4 green pipelines in a row, so submitting this for review now.

### CODE REVIEW

**Comment by mchalski** [comment]
pipelines:
https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/248648755 (fail TestEntMigration)

**Comment by mchalski** [comment]
after refact:
all green https://gitlab.com/Northern.tech/Mender/mender-qa/-/jobs/998543670

**Review by alfrunes — APPROVED** [review]

**Comment by alfrunes** [comment]
You may need to change the `wait_for_traefik` logic to continue retrying if it receives a status code >= 400.

**Review by tranchitella — APPROVED** [review]

**Review by lluiscampos — APPROVED** [review]
Very good commit messages, thanks!

File: testutils/infra/container_manager/docker_compose_manager.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import filelock
 import logging
 import copy
+import redo
```
[the remark this answers is not in the document]
Comment (mchalski) [reply testutils/infra/container_manager/docker_compose_manager.py:22]: thanks!
also - redo is already there! :)
https://github.com/mendersoftware/integration/blob/master/tests/requirements/python-requirements.txt#L21

**Pull request merged**